Obsidian/ChunkData/ChunkSection.cs

```diff
@@ -1,50 +1,36 @@
-﻿using Obsidian.Utilities.Collection;
+﻿using System.Diagnostics;
 
 namespace Obsidian.ChunkData;
 
-public class ChunkSection : BlockStateContainer
+public sealed class ChunkSection
 {
     public bool Overworld = true;
 
     public int? YBase { get; }
-    public override byte BitsPerBlock { get; }
-    public override DataArray BlockStorage { get; }
 
-    public override IBlockStatePalette Palette { get; internal set; }
+    public BlockStateContainer BlockStateContainer { get; }
+    public BiomeContainer BiomeContainer { get; }
 
-    public ChunkSection(byte bitsPerBlock = 6, int? yBase = null)
+    public ChunkSection(byte bitsPerBlock = 4, byte bitsPerBiome = 2, int? yBase = null)
     {
-        this.BitsPerBlock = bitsPerBlock;
-
-        this.BlockStorage = new DataArray(bitsPerBlock, 4096);
-
-        this.Palette = bitsPerBlock.DeterminePalette();
+        this.BlockStateContainer = new(bitsPerBlock);
+        this.BiomeContainer = new(bitsPerBiome);
 
         this.YBase = yBase;
 
-        this.FillWithAir();
+        int airIndex = BlockStateContainer.Palette.GetOrAddId(Block.Air);
+        Debug.Assert(airIndex == 0);
     }
 
     public Block GetBlock(Vector position) => this.GetBlock(position.X, position.Y, position.Z);
-    public Block GetBlock(int x, int y, int z) => this.Get(x, y, z);
+    public Block GetBlock(int x, int y, int z) => this.BlockStateContainer.Get(x, y, z);
 
-    public bool SetBlock(Vector position, Block block) => this.SetBlock(position.X, position.Y, position.Z, block);
-    public bool SetBlock(int x, int y, int z, Block block) => this.Set(x, y, z, block);
+    public Biomes GetBiome(Vector position) => this.GetBiome(position.X, position.Y, position.Z);
+    public Biomes GetBiome(int x, int y, int z) => this.BiomeContainer.Get(x, y, z);
 
-    public bool IsEmpty => this.BlockStorage.Storage.Length <= 0;
+    public void SetBlock(Vector position, Block block) => this.SetBlock(position.X, position.Y, position.Z, block);
+    public void SetBlock(int x, int y, int z, Block block) => this.BlockStateContainer.Set(x, y, z, block);
 
-    private void FillWithAir()
```

Comment:
Is there a specific reason this is removed? e.g. some empty value being interpreted as air? Will accept regardless, just curious. Maybe it was just not needed anymore

Reply:
With the most recent change we made now new palettes are set with air by default so that method wasn't needed anymore